pkgs/development/python-modules/pymilter/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, python, buildPythonPackage, fetchFromGitHub, libmilter, bsddb3, pydns }:
+
+buildPythonPackage rec {
+  pname = "pymilter";
+  version = "1.0.5";
+
+  src = fetchFromGitHub {
+    owner = "sdgathman";
+    repo = pname;
+    rev = "${pname}-${version}";
+    hash = "sha256-gZUWEDVZfDRiOOdG3lpiQldHxm/93l8qYVOHOEpHhzQ=";
+  };
+
+  buildInputs = [ libmilter ];
+  propagatedBuildInputs = [ bsddb3 pydns ];
+
+  preBuild = ''
+    sed -i 's/import thread/import _thread as thread/' Milter/greylist.py
+  '';
+
+  # requires /etc/resolv.conf
+  doCheck = false;
```

Comment:
We should at least have a `pythonImportsCheck` if we're not going to have the tests enabled.

FWIW see the `libredirect` package and its various uses around nixpkgs for how to provide "fake" files at arbitrary filesystem locations for test suites.